x/sequencers/keeper/dymint.go

```diff
@@ -13,40 +13,59 @@ import (
 )
 
 // set dymint sequencers from InitChain
-func (k Keeper) SetDymintSequencers(ctx sdk.Context, validators []abci.ValidatorUpdate) {
-	for _, val := range validators {
-		tmkey, err := tmcrypto.PubKeyFromProto(val.PubKey)
+func (k Keeper) SetDymintSequencers(ctx sdk.Context, sequencers []abci.ValidatorUpdate) {
+	if len(sequencers) == 0 {
+		panic(types.ErrNoSequencerOnInitChain)
+	}
+
+	if len(sequencers) > 2 {
+		panic(types.ErrMultipleDymintSequencers)
+	}
+
+	var (
+		operatorAddr    sdk.ValAddress
+		power           int64
+		consensusPubKey cryptotypes.PubKey
+		operatorPubkey  cryptotypes.PubKey
+	)
+
+	for _, seq := range sequencers {
+		tmkey, err := tmcrypto.PubKeyFromProto(seq.PubKey)
 		if err != nil {
 			panic(err)
 		}
 		pubKey, err := cryptocodec.FromTmPubKeyInterface(tmkey)
 		if err != nil {
 			panic(err)
```

Comment:
Not sure I understand this logic/use-case. what are we trying to achieve here? 

also if the pubKey != ed25519 it will always panic later so why we go to the else clause?

Reply:
I think it's fine, the pub key can also be secp for the operator